qp_deblur/deblur.py

```diff
@@ -157,6 +159,177 @@ def generate_sepp_placements(seqs, out_dir, threads, reference_phylogeny=None,
         raise ValueError(error_msg)
 
 
+def _get_guppy_binary():
+    """Find the right guppy binary.
+
+    Returns
+    -------
+    str : filepath of the guppy binary.
+
+    Raises
+    ------
+    ValueError if
+        a) grep'ing DIR in the run-sepp.sh script or
+        b) grep'ing pplacer path in main.config fails
+    """
+    # 1) locate sepp binary
+    fp_sepp_binary = which('run-sepp.sh')
+    # 2) obtain shared directory by grep'ing DIR from the run-sepp.sh script
+    std_out, _, return_value = system_call('grep "^DIR=" %s' % fp_sepp_binary)
+    if return_value != 0:
+        raise ValueError("Could not determine SEPP shared directory.")
+    fp_sepp_shared = std_out.split('=')[-1].strip()
+    # 3) grep'ing pplacer path
+    std_out, _, return_value = system_call(
+        'grep "\[pplacer\]" %s -A 1 | grep "path"' %
+        join(fp_sepp_shared, 'sepp', '.sepp', 'main.config'))
+    if return_value != 0:
+        raise ValueError("Could not determine SEPP bundled tool directory.")
+    fp_sepp_bundled = split(std_out.split('=')[-1].strip())[0]
+    return join(fp_sepp_bundled, 'guppy')
+
+
+def _generate_template_rename(file_reference_phylogeny,
+                              file_reference_alignment,
+                              out_dir):
+    """Produces placement template and rename script for reference phylogeny.
+
+    Parameters
+    ----------
+    file_reference_phylogeny : str
+        A filepath to an alternative reference phylogeny for SEPP.
+    file_reference_alignment : str
+        A filepath to an alternative reference alignment for SEPP.
+    out_dir : str
+        The job output directory
+
+    Returns
+    -------
+    (str, str) : Filepaths of reference_template json file and
+    reference_rename python script.
+
+    Notes
+    -----
+    This function only needs to be called once per reference phylogeny/
+    alignment, i.e. if we update Greengenes or extend SEPP for Silva or other
+    reference phylogenies. I am including this function for easier maintainance
+    in the future.
```

Comment:
I foresee very few updates. How often has Greengenes updated in the last years? Hopefully we can add Silva soon.
I also thought about this option, but I decided to go with "static" files that are relocated by the setup.py script instead of dynamically generating those because it would impose some overhead since we for sure will "install" the plugin way more often than updating any references.

Reply:
I think this points out that we will need some kind of function to "initialize" and archive, so this function can be called when a new archive is created. I think, however, that's a different discussion out of the scope of this PR.